include/deal.II/grid/grid_tools.h

```diff
@@ -974,6 +974,41 @@ namespace GridTools
   get_patch_around_cell(const typename MeshType::active_cell_iterator &cell);
 
 
+  /**
+   * This function runs through the degrees of freedom defined by the
+   * Container and for each dof constructs a vector of active_cell_iterators
+   * representing the cells of support of the nodal basis element
+   * at that degree of freedom. This function was designed for the
+   * implementation of local projections, for instance the Clement interpolant,
+   * in conjunction with other local patch functions like
+   * GridTools::build_triangulation_from_patch.
+   *
+   * Containers built on top of Triangulation<dim> or
+   * parallel:distributed::Triangulation<dim> are supported and handled
+   * appropriately.
+   *
+   * It is necessary that the finite element underlying the Container used has
+   * degrees of freedom on faces (2d or 3d) and lines (in 3d).  This unfortunately
+   * precludes the FE_DGQ<dim> finite element.  Likewise, the finite element
```

Comment:
I suppose that in the strict sense of the clement operator, for a DG element, the patch for each DoF would be the cell to which it is associated.  I guess in the practice we often think of the clement operator as taking a discontinuous function and returning a continuous function but that is not always true.  Only if the underlying finite element is continuous.  

By Q1 x DGQ1 you mean a vector finite element with components in those various spaces?  Yeah, that might be an interesting case...  I would probably say that it should be decomposed into patches for each component of a vector finite element.  Maybe not the most efficient but it would be clear how to do it.